src/OrchardCore.Modules/OrchardCore.Forms/Services/ValidationRuleServiceExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Forms.Providers;
+
+namespace OrchardCore.Forms.Services
+{
+    public static class ValidationRuleServiceExtensions
+    {
+        /// <summary>
+        /// Adds validation rule services.
+        /// </summary>
+        public static IServiceCollection AddValidationRules(this IServiceCollection services)
+        {
+            services.AddScoped<IValidationRuleService, ValidationRuleService>();
```

Comment:
It might be useful to create a custom content item metadata for things that can be validated by returning their value in specific logical types, and each validator could handle these special logical types (datetime, numeric, text, boolean, json, html, any). Then a form widget could implement this metadata. However I haven't looked an how validators know which property then need to validate for a specific widget, so I might be mislead.